application-diagram-api/src/main/java/com/xwiki/diagram/internal/DiagramMacroRunnable.java

```diff
@@ -0,0 +1,143 @@
+ */
+package com.xwiki.diagram.internal;
+
+import java.util.List;
+import java.util.concurrent.BlockingQueue;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+
+import org.slf4j.Logger;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.model.reference.EntityReferenceSerializer;
+import org.xwiki.rendering.block.Block;
+import org.xwiki.rendering.block.Block.Axes;
+import org.xwiki.rendering.block.XDOM;
+import org.xwiki.rendering.block.match.MacroBlockMatcher;
+
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.XWikiException;
+import com.xpn.xwiki.doc.XWikiDocument;
+
+/**
+ * Updates diagram macro reference parameter after the rename of a diagram.
+ * 
+ * @version $Id$
+ * @since 1.14
+ */
+@Component(roles = DiagramMacroRunnable.class)
+@Singleton
+public class DiagramMacroRunnable extends AbstractDiagramRunnable
+{
+    private static final String MACRO_REFERENCE_PARAMETER = "reference";
+    @Inject
+    private Provider<XWikiContext> contextProvider;
+
+    @Inject
+    @Named("compact")
+    private EntityReferenceSerializer<String> compactEntityReferenceSerializer;
+
+    @Inject
+    private Logger logger;
+
+    /**
+     * @see com.xpn.xwiki.util.AbstractXWikiRunnable#runInternal()
+     */
+    @Override
+    public void runInternal()
+    {
+        while (!Thread.interrupted()) {
+            DiagramQueueEntry queueEntry;
+            BlockingQueue<DiagramQueueEntry> diagramsQueue = this.getDiagramsQueue();
+
+            try {
+                queueEntry = diagramsQueue.take();
+            } catch (InterruptedException e) {
+                logger.warn("Diagrams update thread has been interrupted", e);
+                queueEntry = STOP_RUNNABLE_ENTRY;
+            }
+
+            if (queueEntry == STOP_RUNNABLE_ENTRY) {
+                diagramsQueue.clear();
+                break;
+            }
+
+            XWikiContext xcontext = contextProvider.get();
+            DocumentReference originalDocRef = queueEntry.originalDocRef;
+            DocumentReference currentDocRef = queueEntry.currentDocRef;
+
+            try {
+                // We need to take backlinks from the original document because at this step they are not loaded on
+                // the new document.
+                List<DocumentReference> backlinks =
+                    xcontext.getWiki().getDocument(originalDocRef, xcontext).getBackLinkedReferences(xcontext);
+
+                for (DocumentReference backlinkDocRef : backlinks) {
+                    XWikiDocument backlinkDoc = xcontext.getWiki().getDocument(backlinkDocRef, xcontext);
+
+                    updateDiagramMacrosReferences(backlinkDoc, currentDocRef, originalDocRef, xcontext);
+                }
+            } catch (XWikiException e) {
+                logger.debug(e.getFullMessage());
```

Comment:
Are you sure the error message should be visible only for debug log level?

Reply:
I forgot to change that :)